WebCryptoAPI/sign_verify/ecdsa.js

```diff
@@ -0,0 +1,400 @@
+                });
+            }, vector.name + " using publicKey to sign");
+        }, function(err) {
+            promise_test(function(test) {
+                assert_unreached("importVectorKeys failed for " + vector.name + ". Message: ''" + err.message + "''");
+            }, "importVectorKeys step: " + vector.name + " using publicKey to sign");
+        });
+
+        all_promises.push(promise);
+    });
+
+    // Check for failures due to no "verify" usage.
+    testVectors.forEach(function(originalVector) {
+        var vector = Object.assign({}, originalVector);
+
+        var promise = importVectorKeys(vector, [], ["sign"])
+        .then(function(vectors) {
+            var algorithm = {name: vector.algorithmName, hash: vector.hashName};
+            promise_test(function(test) {
+                return subtle.verify(algorithm, vector.publicKey, vector.signature, vector.plaintext)
+                .then(function(plaintext) {
+                    assert_unreached("Should have thrown error for no verify usage in " + vector.name + ": " + err.message + "'");
+                }, function(err) {
+                    assert_equals(err.name, "InvalidAccessError", "Should throw InvalidAccessError instead of '" + err.message + "'");
+                });
+            }, vector.name + " no verify usage");
+        }, function(err) {
+            promise_test(function(test) {
+                assert_unreached("importVectorKeys failed for " + vector.name + ". Message: ''" + err.message + "''");
+            }, "importVectorKeys step: " + vector.name + " no verify usage");
+        });
+
+        all_promises.push(promise);
+    });
+
+    // Check for successful signing and verification.
+    testVectors.forEach(function(vector) {
+        var promise = importVectorKeys(vector, ["verify"], ["sign"])
+        .then(function(vectors) {
+            var algorithm = {name: vector.algorithmName, hash: vector.hashName};
+            promise_test(function(test) {
+                return subtle.sign(algorithm, vector.privateKey, vector.plaintext)
+                .then(function(signature) {
+                    // Can we get verify the signature?
+                    return subtle.verify(algorithm, vector.publicKey, signature, vector.plaintext)
+                    .then(function(is_verified) {
+                        assert_true(is_verified, "Round trip verification works");
+                        return signature;
+                    }, function(err) {
+                        assert_unreached("verify error for test " + vector.name + ": " + err.message + "'");
+                    });
+                })
+                .then(function(priorSignature) {
+                    // Will a second signing give us different signature, as it should?
+                    return subtle.sign(algorithm, vector.privateKey, vector.plaintext)
```

Comment:
I'm fine removing those tests, but I don't think it's needed. The
recommendation specifically says that those curves should follow X96.2,
which is older than this RFC. I can't read that spec, though, since it's
for sale, not open access. So I don't know whether it would allow the
non-random solution given in the RFC or not.

I'll let you choose. Let me know whether to pull that set of tests out.

Thanks.

On Fri, Jun 17, 2016 at 3:50 AM, Jim Schaad <notifications@github.com>
wrote:

> In WebCryptoAPI/sign_verify/ecdsa.js
> <https://github.com/w3c/web-platform-tests/pull/3183#discussion_r67471928>
> :
>
> > +            var algorithm = {name: vector.algorithmName, hash: vector.hashName};
> > +            promise_test(function(test) {
> > +                return subtle.sign(algorithm, vector.privateKey, vector.plaintext)
> > +                .then(function(signature) {
> > +                    // Can we get verify the signature?
> > +                    return subtle.verify(algorithm, vector.publicKey, signature, vector.plaintext)
> > +                    .then(function(is_verified) {
> > +                        assert_true(is_verified, "Round trip verification works");
> > +                        return signature;
> > +                    }, function(err) {
> > +                        assert_unreached("verify error for test " + vector.name + ": " + err.message + "'");
> > +                    });
> > +                })
> > +                .then(function(priorSignature) {
> > +                    // Will a second signing give us different signature, as it should?
> > +                    return subtle.sign(algorithm, vector.privateKey, vector.plaintext)
>
> I do not believe that this is a correct text. See RFC 6979 for a
> deterministic version of ECDSA. This is the trend for signature algorithms
> as we go forward as it removes a major security issue with using the same
> value of k for two different messages.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/w3c/web-platform-tests/pull/3183/files/7f598aa6098a188859bea290f147c48b5712e999#r67471928>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAE_N_T7cQxtZrogQuSyvrLpcNYCMuGQks5qMlHcgaJpZM4I2ltT>
> .
>